pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -116,6 +116,25 @@ func (r *ReconcileMasterUserRecord) Reconcile(request reconcile.Request) (reconc
 			logger.Error(err, "unable to add finalizer to MasterUserRecord")
 			return reconcile.Result{}, err
 		}
+		// migration for CRT-1075: add an annotation with the email address (same as for associated UserSignup resource)
+		if mur.Annotations == nil || mur.Annotations[toolchainv1alpha1.MasterUserRecordEmailAnnotationKey] == "" {
+			usersignup := &toolchainv1alpha1.UserSignup{}
+			if r.client.Get(context.TODO(), types.NamespacedName{
+				Namespace: mur.Namespace,
+				Name:      mur.Spec.UserID,
```

Comment:
`UserID` doesn't necessarily match `UserSignup.Name`.
It used to but not anymore. We have to get the usersignup name from either `toolchain.dev.openshift.com/owner` label or `ownerReference`.

Reply:
Actually, `MUR.Spec.UserID` is actually the corresponding `UserSignup.Name`. The UserID name of this field is confusing. We even have a JIRA to rename it: https://issues.redhat.com/browse/CRT-942
So, While actual UserID is not longer necessarily the same as the UserSignup name (atm for most resources it's the same but for some it's not) this particular field refers to UserSignup.Name.